examples/getting_started/dynamic_config_and_chaining.ipynb

```diff
@@ -0,0 +1,734 @@
+    "\n",
+    "\n",
+    "# Low creativity recommendation\n",
+    "response = recommend_book(\"mystery\", 0.2)\n",
+    "print(\"Low creativity:\", response.content)\n",
+    "\n",
+    "# High creativity recommendation\n",
+    "response = recommend_book(\"mystery\", 0.8)\n",
+    "print(\"High creativity:\", response.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Computed Fields\n",
+    "\n",
+    "Computed fields allow you to dynamically generate or modify template variables used in your prompt. For more information on prompt templates, see the [Prompts documentation](https://docs.mirascope.io/learn/prompts/)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-10T04:44:12.163281Z",
+     "start_time": "2024-09-10T04:44:11.065965Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "I recommend **\"The Cruel Prince\" by Holly Black**. It's a captivating young adult fantasy that follows Jude, a mortal girl living in the treacherous world of Faerie. The story is filled with political intrigue, betrayal, and magic, as Jude navigates her place among the powerful fae, all while trying to prove herself worthy of a role in their world. The book is the first in \"The Folk of the Air\" trilogy, offering an engaging blend of adventure and romance that appeals to young adult readers.\n"
+     ]
+    }
+   ],
+   "source": [
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Recommend a {genre} book with a reading level of {reading_level}\")\n",
+    "def recommend_book_by_age(genre: str, age: int) -> openai.OpenAIDynamicConfig:\n",
+    "    reading_level = \"adult\"\n",
+    "    if age < 12:\n",
+    "        reading_level = \"elementary\"\n",
+    "    elif age < 18:\n",
+    "        reading_level = \"young adult\"\n",
+    "    return {\"computed_fields\": {\"reading_level\": reading_level}}\n",
+    "\n",
+    "\n",
+    "response = recommend_book_by_age(\"fantasy\", 15)\n",
+    "print(response.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Dynamic Tools\n",
+    "\n",
+    "You can dynamically configure which tools are available to the LLM based on runtime conditions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-10T04:44:16.869524Z",
+     "start_time": "2024-09-10T04:44:15.618839Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The Girl with the Dragon Tattoo by Stieg Larsson (mystery)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from mirascope.core import BaseToolKit, toolkit_tool\n",
+    "\n",
+    "\n",
+    "class BookToolkit(BaseToolKit):\n",
```

Comment:
nit: `BaseToolKit` might be a little overkill for this getting started guide. Maybe just show return a basic tool function here? and then add a link to the toolkit docs and/or the tools + agents getting started guide saying that that's where this is most useful?